### DISCUSSION TITLE
Sonoff ZigBee Bridge Pro - Tasmota Script missing

### DISCUSSION THREAD

**Discussion by 2000ede** [arendst/Tasmota#22341] [Support request] [answered]
![grafik](https://github.com/user-attachments/assets/523c3ba8-7664-470a-8ee9-107f334c3bc2)

Hello,

I have successfully flashed the Sonoff ZigBee Bridge Pro with Tasmota:
my Help: https://atlane.de/sonoff-zigbee-bridge-pro-tasmota-flashing-und-homeassistant-integration/

Unfortunately I am missing the script function (in the bin file).

What special defines do I need for the bridge?

With my standard bin he says there is not enough space. 
You may have to use the correct bin from the start.

Thanks for the help!

https://templates.blakadder.com/sonoff_ZBBridge-P.html
https://zigbee.blakadder.com/Sonoff_ZBBridge-P.html

**Comment by Jason2866** [comment]
The build variant does include rules. what is your use case for scripting on this device?

**Reply by 2000ede** [reply]
Hello @Jason2866 , 

thank you for your answer.

I'm not a trained developer, but I've already familiarized myself with scripts (with help) and have implemented a lot with them. I know my stuff there well and I also like script.
I've never used rules or the syntax (or berry). I'm afraid it would take up a lot of my time.

My use case is, among other things, to make senior buttons. When my parents press the smart button I want to receive a Telegram message.

Sorry for my english. Greetings from Germany!

**Reply by Jason2866** [reply]
Especially with this device is not a good idea to hook up additional stuff. The zigbee part should be not "disturbed" by other tasks. To keep stuff stable better use a additional device.
Anyways you have to compile your build yourself. The predefined build variant allows to change from rules to scripter. So do as described in Tasmota documentation

**Reply by 2000ede** [reply]
(what do you mean by additional devices? I just want to connect the Tuya Zigbee devices (e.g. Smart Button). But then just send something to Telegram via https.)

I'm already creating bin files for other projects. But here I just don't know what I need for **Define for this special device (Gateway)**. And there seems to be more in the bin here.
That's why I'm asking specifically here.

**Reply by Jason2866** [reply]
Just undef rules and define scripter. As said before this special env allows this change. Nothing else

**Reply by 2000ede** [reply]
So far I only had one finished bin file. And I don't know what special ingredients are included:
- define for the gateway
- platform override 
- special files

But ok, I will now try it with a normal ESP32 bin file including script. I'll be surprised if that works.

**Reply by sfromis** [reply]
The point was that you can take the existing build environment for the Sonoff Zigbee Bridge "pro", and then make the mentioned tweaks, instead of starting from "a normal ESP32".

Or you could use the Berry language instead of the scripter, but that's a quite different thing...

**Reply by 2000ede** [reply]
@sfromis 

I apologize, maybe it's because of my English.

That's exactly what I'm looking for: 
Existing base files/version for the Sonoff Zigbee Bridge “pro”

So far I have only found finished bin files. And I can't take them apart.

Where exactly can I find it?
existing build environment for the Sonoff Zigbee Bridge "pro" ?

**Reply by sfromis** [reply, marked as answer]
For custom build, see https://tasmota.github.io/docs/Compile-your-build/
And the mentioned build env: https://github.com/arendst/Tasmota/blob/development/platformio_tasmota_env32.ini#L261

**Reply by 2000ede** [reply]
@sfromis 
thanks, that was what I was looking for. I wasn't aware that it was simply a module selection in the platform override.

I was able to create and install the bin cleanly.

Unfortunately it doesn't work like it did with the original bin:
https://atlane.de/sonoff-zigbee-bridge-pro-tasmota-flashing-und-homeassistant-integration/

I have saved the profile first:
{"NAME":"TCP ZBBridge Pro","GPIO":[0,0,576,0,480,0,0,0,0,1,1,5792,0,0,0,5472,0,320,5793,5504,0,640,608,32,0,0,0,0,0,1,0,0,0,0,0,0],"FLAG":0,"BASE":1}


After I installed bin, I entered this into Berry and the response was as expected:

import sonoff_zb_pro_flasher as cc
cc.load("SonoffZBPro_coord_20220219.hex")
cc.check()

Then I entered the following into Berry, and it also confirmed this correctly in the log:
cc.flash()

After the restart, he didn't create the zigbee buttons on the Tastmota page and the console didn't show what he expected:

my output:
12:08:25.655 RSL: INFO1 = {"Info1":{"Module":"TCP ZBBridge Pro","Version":"14.2.0.4(efGM9924)","FallbackTopic":"cmnd/DVES_575FCC_fb/","GroupTopic":"cmnd/tasmotas/"}}
12:08:25.669 RSL: INFO2 = {"Info2":{"WebServerMode":"Admin","Hostname":"tasmota-575FCC-8140","IPAddress":"192.168.178.199","IP6Global":"2a02:8206:a6b5:8300:ca2e:18ff:fe57:5fcc","IP6Local":"fe80::ca2e:18ff:fe57:5fcc%st1"}}
12:08:25.681 RSL: INFO3 = {"Info3":{"RestartReason":"Vbat power on reset","BootCount":6}}
12:08:26.933 QPC: Reset
12:08:29.958 RSL: STATE = {"Time":"2024-10-24T12:08:29","Uptime":"0T00:00:10","UptimeSec":10,"Heap":106,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":0,"Berry":{"HeapUsed":3,"Objects":38},"Wifi":{"AP":1,"SSId":"Fichtenstr9","BSSId":"5C:49:79:CA:9B:D8","Channel":1,"Mode":"HT20","RSSI":90,"Signal":-55,"LinkCount":1,"Downtime":"0T00:00:05"}}

expected according to the instructions:
20:57:28.054 ZIG: rebooting ZNP device
20:57:28.972 ZbInput discarding byte 00
20:57:28.976 RSL: RESULT = {"ZbState":{"Status":1,"Message":"CCxxxx ZNP booted","RestartReason":"Power-up","MajorRel":2,"MinorRel":7}}
20:57:29.178 RSL: RESULT = {"ZbState":{"Status":50,"MajorRel":2,"MinorRel":7,"MaintRel":1,"Revision":20220219}}
20:57:29.229 RSL: RESULT = {"ZbState":{"Status":2,"Message":"Resetting configuration"}}
20:57:30.859 ZbInput discarding byte 00
20:57:31.429 RSL: RESULT = {"ZbState":{"Status":3,"Message":"Configured, starting coordinator"}}
20:57:37.343 RSL: RESULT = {"ZbState":{"Status":40,"NewState":9,"Message":"Started as coordinator"}}
20:57:37.446 RSL: RESULT = {"ZbState":{"Status":51,"IEEEAddr":"0x00XXXXXXXXXX","ShortAddr":"0x0000","DeviceType":7,"DeviceState":9,"NumAssocDevices":0}}
20:57:37.957 RSL: RESULT = {"ZbState":{"Status":0,"Message":"Started"}}
20:57:37.960 ZIG: Zigbee started
20:57:37.980 ZIG: No Zigbee device information
20:57:37.990 ZIG: No Zigbee device data

With the original bin it worked perfectly. where can I find my error?

**Reply by Jason2866** [reply]
Never build this variant without rules. It could be it does not work without

**Reply by 2000ede** [reply]
@sfromis @Jason2866 
thanks for you support!
Now it works, I took a different guide to parameterizing Tasmota. Instead of the Berry inputs I did this and it works! SUPER

https://templates.blakadder.com/sonoff_ZBBridge-P.html

-->
Auto-configuration
After flashing Tasmota, open the web UI of the device and navigate to Configuration -> Auto-configuration. Select your device from the drop-down and click Apply Configuration.
The device will restart and will be automagically configured.

**Reply by sfromis** [reply]
Well, when you set up to use the TCP setup for Z2M or ZHA, you'll not get **any** Zigbee stuff in Tasmota log or web UI, as **everything** happens on the other end. And it is not really relevant to fuss about using the scripter, when you do not do anything on the Tasmota device outside of providing the ser2net tunnel (where you need a `TCPStart` command to open for connections).

**Reply by 2000ede** [reply]
I would like, among other things,

My parents press a smart button and I get a Telegram message.

I already do this with other Tasmota projects (solar monitoring).

**Reply by sfromis** [reply]
What I meant was that if/when you set up for Zigbee via the ser2net TCP tunnel, you can do nothing Zigbee-related on the Tasmota device as everything is handled on the other end. Hence having Berry or scripter code on the bridge does not make sense.
